### PR TITLE
add win32 egl gpu blacklist.

### PR DESCRIPTION
**Pull request by danwalmsley**
## What does the pull request do?

Adds a blacklist to prevent "Microsoft Basic Renderer" software rasterizer being used.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0019836-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by kekekeks — APPROVED** [review]

File: src/Windows/Avalonia.Win32/Win32Platform.cs [hunk 1]
```diff
@@ -56,6 +56,11 @@ public class Win32PlatformOptions
         /// GPU rendering will not be enabled if this is set to false.
         /// </remarks>
         public bool? AllowEglInitialization { get; set; }
+        
+        public IList<string> EglRendererBlacklist { get; set; } = new List<string>
```
Comment (workgroupengineering) [inline src/Windows/Avalonia.Win32/Win32Platform.cs:60]: IReadOnlyList<T>?

Comment (jmacato) [reply src/Windows/Avalonia.Win32/Win32Platform.cs:60]: it could be useful to add one in runtime?

Comment (workgroupengineering) [reply src/Windows/Avalonia.Win32/Win32Platform.cs:60]: I think at runtime it is not useful. If I need to add a render to the blaklist, I will always have to restart the app to make it active.

Comment (workgroupengineering) [reply src/Windows/Avalonia.Win32/Win32Platform.cs:60]: I think at runtime it is not useful. If I need to add a render to the blaklist, I will always have to restart the app to make it active.

File: src/Windows/Avalonia.Win32/Win32Platform.cs [hunk 2]
```diff
@@ -56,6 +56,11 @@ public class Win32PlatformOptions
...
+        {
+            "Microsoft Basic Renderer"
```
Comment (wieslawsoltes) [inline src/Windows/Avalonia.Win32/Win32Platform.cs]: "Microsoft Basic Render"

**Review by wieslawsoltes — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0019886-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Pull request merged**